relayer/src/ethereum_checkpoints/sync_update.rs

```diff
@@ -0,0 +1,137 @@
+use super::*;
+use anyhow::{anyhow, Result as AnyResult};
+pub use checkpoint_light_client_io::sync_update::Error;
+use std::ops::ControlFlow::{self, *};
+
+pub fn spawn_receiver(
+    client_http: Client,
+    beacon_endpoint: String,
+    sender: Sender<SyncCommitteeUpdate>,
+    delay: Duration,
+) {
+    tokio::spawn(async move {
+        log::info!("Update receiver spawned");
+
+        let mut failures = 0;
+        loop {
+            match receive(
+                &client_http,
+                &beacon_endpoint,
+                &sender,
+                delay,
+                &mut failures,
+            )
+            .await
+            {
+                Break(_) => break,
+                Continue(_) => (),
+            }
+        }
+    });
+}
+
+async fn receive(
+    client_http: &Client,
+    beacon_endpoint: &str,
+    sender: &Sender<SyncCommitteeUpdate>,
+    delay: Duration,
+    failures: &mut usize,
+) -> ControlFlow<()> {
+    // helper macro
+    macro_rules! check_and_return {
+        () => {
+            *failures += 1;
+            if *failures >= COUNT_FAILURE {
+                return Break(());
+            }
+
+            time::sleep(delay).await;
+
+            return Continue(());
+        };
+    }
```

Comment:
Why you don't just return Result<ControlFlow<()>> and check failure count outside? It will reduce boilerplate to `?`